include/sys/time_units.h

```diff
@@ -11,6 +11,13 @@
 extern "C" {
 #endif
 
+/** @brief System-wide macro to denote "forever" in milliseconds
+ *
+ *  Usage of this macro is limited to APIs that want to expose a timeout value
+ *  that can optionally be unlimited, or "forever".
+ */
+#define SYS_FOREVER_MS -1
```

Comment:
It's not this simple, because now K_MSEC(SYS_FOREVER_MS) produces some crazy value and not K_FOREVER, which is what you would want.  The Z_TIMEOUT_MS() constructor macro would have to add a test for this.

(Then someone would come along and ask why we don't have SYS_FOREVER_US and SYS_FOREVER_NS, nor SYS_FOREVER_TICKS for that matter even though ticks of -1 is how it works internally).

Reply:
@andyross Sure, so I would need to explicitly document that this is not compatible with `K_MSEC()`, which is fair since it's a `SYS_` macro not a `K_` one.

Reply:
I've added a comment and a `SYS_MSEC` macro to clarify this @andyross